mathematical-companion/main.tex

```diff
@@ -0,0 +1,1300 @@
+
+Suppose that a user's share data has up to 8 errors in it. We can write their data
+in polynomial form as
+\[ \hat{d}(x) = d(x) + e(x) \]
+where $d(x)$ is their actual share data, $e(x)$ is some error polynomial with up
+to 8 terms, and $\hat{d}$ is their actual data.
+
+Taking this modulo $G$, we see that
+\[ \hat{d}(x) \equiv d(x) + e(x) \equiv \vc{SECRETSHARE32} + e(x) \mod G. \]
+That is, in the process of modding out by $G$, the actual secret data is completely
+erased and is replaced by the target residue. This is why the booklet advises users
+that they may enter the residue into an electronic computer --- it has no secret
+data in it --- and why this is a useful thing to do --- it is completely determined
+by the error pattern.
+
+The task of \textbf{error correction} is to undo the reduction process. That is,
+given a residue $e(x)$ mod $G$, figure out what $e(x)$ really is. Then the user
+can recover their share data by computing
+\[ \hat{d}(x) + e(x) = d(x) + e(x) + e(e) = d(x) \]
+
+Because our checksum has distance 9, if there are up to 8 errors, the reduction
+is guaranteed not to produce the value \vc{SECRETSHARE32}, so that the error
+pattern will be recognized as an error.
+
+Furthermore, in all cases where there are 4 or fewer errors, the reduction will
+produce unique values, so that it can (in principle) be undone.
+
+But doing this is fairly involved. The standard way of doing this is to use the
+Berlekamp-Massey algorithm to determine an ``error locator polynomial'', to
+find roots of this polynomial to determine the error locations, and then to use
+Forney's algorithm to compute the error values. All of these steps involve
+arithmetic in $\ftttwo$ which does not lend itself nicely to volvelles and
+worksheets. We believe that it possible, but as of March 2023, we have not
+figured out how.
+
+However, one thing we \emph{can} do is produce a lookup table. In the case that
+the user has a 48-element share, of which the first 3 elements are definitely
+the characters \vc{ms1}, then there are 1395 ways in which there may be one
+mistake in the remaining 45 characters. Each of these 1395 ways produce a unique
+residue, and we have provided a 3-page table listing them all. Therefore, if
+the user makes only a single error, they can simply look up their residue to
+learn how to correct it.
+
+A final observation is that errors might not mean that the share data itself are
+corrupted. An error in location $y$ actually means that \emph{column $y$ in the
+Checksum Worksheet was not computed correctly}. This may mean that the top cell,
+which contains share data, was wrong. But it could also be that the user made an
+arithmetic error. We therefore advise users, when doing error correction, to
+completely recompute any erroneous columns of the Checksum Worksheet before they
+consider modifying their data.
+
+\section{Conclusion and Acknowledgements}
+
+We thank the authors of SLIP39 for noticing the remarkable compatibility between
```

Comment:
AFAIK, the authors of SLIP39 didn't notice this compatibility. It was Leon.

Reply:
Ok, I'll update this text.